app/controllers/broadcasts_controller.rb

```diff
@@ -41,17 +42,14 @@ def new_with_users
 
     @broadcast = Broadcast.accessible_by(current_ability).new
     users = User.accessible_by(current_ability).with_assoc.by_name
+    users = apply_search_if_given(Question, users)
+    users = restrict_scope_to_selected_objects(users)
+    @broadcast.recipient_users = users
```

Comment:
In the old code we don't assign users if it's the 'all users' case. Can you please fix to match that please. In general I am not super confident all the different combinations are correctly treated here. It's kind of delicate. Can you take another look please?

Reply:
Also can you please confirm separately that 

1. if I have one _unfiltered_ page of e.g. 10 users and I click 'Select All' and then send broadcast, the params[:select_all] will be set?
2. If I have one _filtered_ page of users and I click 'Select All' and then send broadcast, select_all _won't_ be set?

Or I'm not even sure this is the right approach.

I am worried we are going too fast here and we need to be careful about this. It's tricky. This is kind of why I wanted to rename the `select_all` param. We need to be crystal clear about what it means.

Can you please make up a truth table of all the possible combinations here and let's review it together?